pkg/services/folder/folderimpl/folder.go

```diff
@@ -115,6 +115,10 @@ func (s *Service) Get(ctx context.Context, cmd *folder.GetFolderQuery) (*folder.
 		return nil, folder.ErrBadRequest.Errorf("missing signed in user")
 	}
 
+	if cmd.UID != nil && *cmd.UID == folder.SharedFolderUID {
```

Comment:
we could also hide it behind the `nestedFolders` feature flag (no point to return this if it's not enabled)